pkg/scalers/scalersconfig/typed_config.go

```diff
@@ -52,6 +52,8 @@ var allowedParsingOrderMap = map[ParsingOrder]bool{
 	AuthParams:      true,
 }
 
+var elemKeyValSeparator = []string{"=", ":"}
```

Comment:
supporting multiple separators is a great idea but I think it might be safer to have it configurable instead. Having multiple different separators simultaneously, would exclude possible configuration options in the future, for example, if someone decided to put server URLs here, it wouldn't parse the way I would consider intuitive
```
srv1=https://server1:8080,srv2=https://server2:8080
```

It could be configurable with a new field tag parameter, maybe called `separator`, user would have this under better control. Default probably `=`, because that is how the majority of scaler configs seem to be written.

https://github.com/kedacore/keda/blob/00baf2839c438524c487b684bf3b01770d6e9422/pkg/scalers/scalersconfig/typed_config.go#L69-L79

https://github.com/kedacore/keda/blob/00baf2839c438524c487b684bf3b01770d6e9422/pkg/scalers/scalersconfig/typed_config.go#L81-L112

Reply:
Agree with you on this, lets put the custom separator in the TODO list for now